src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,34 +32,45 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        if (address == null) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressDetails = address.split(REGEX_ADDRESS_SPLIT);
+        if (!isValidAddress(addressDetails)) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        block = new Block(addressDetails[DETAIL_INDEX_BLOCK]);
+        street = new Street(addressDetails[DETAIL_INDEX_STREET]);
+        unit = new Unit(addressDetails[DETAIL_INDEX_UNIT]);
+        postalCode = new PostalCode(addressDetails[DETAIL_INDEX_POSTAL_CODE]);
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given address detail array is a valid person address.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] details) {
+        return details.length == SIZE_ADDRESS_DETAIL;
```

Comment:
You can also modify `REGEX_ADDRESS_VALIDATION` to check if the user's input fit your required format other than checking the length.